pkg/ccl/importccl/import_pgdump_test.go

```diff
@@ -0,0 +1,268 @@
+		},
+		{
+			name: "unqualified create statements",
+			data: `
+		CREATE DATABASE db;
+		CREATE SCHEMA s;
+		CREATE TABLE t (id INT);
+		CREATE SEQUENCE seq;
+		CREATE INDEX idx ON t (id);
+		`,
+			expectedStmts: []string{"CREATE SCHEMA s",
+				"CREATE TABLE t (id INT8)",
+				"CREATE SEQUENCE seq",
+				"CREATE INDEX idx ON t (id)"},
+		},
+		{
+			name: "qualified create statement",
+			data: `
+		CREATE DATABASE db;
+		CREATE SCHEMA db.s;
+		CREATE TABLE db.s.t (id INT);
+		CREATE TABLE s.t2 (id INT);
+		CREATE SEQUENCE db.public.seq;
+		CREATE INDEX idx ON db.s.t (id);
+		`,
+			expectedStmts: []string{"CREATE SCHEMA crdb_temp_pgdump_import.s",
+				"CREATE TABLE crdb_temp_pgdump_import.s.t (id INT8)",
+				"CREATE TABLE s.t2 (id INT8)",
+				"CREATE SEQUENCE crdb_temp_pgdump_import.public.seq",
+				"CREATE INDEX idx ON crdb_temp_pgdump_import.s.t (id)"},
+		},
+		{
+			name: "mismatched catalog name",
+			data: `
+		CREATE DATABASE db;
+		CREATE SCHEMA db2.s;
+		`,
+			error: "catalog name db2 does not match dump target database name db",
+		},
+		{
+			name: "unqualified alter statements",
+			data: `
+CREATE DATABASE db;
+ALTER TABLE t ADD COLUMN col STRING;
+ALTER SCHEMA s RENAME TO s2;
+		`,
+			expectedStmts: []string{"ALTER TABLE t ADD COLUMN col STRING",
+				"ALTER SCHEMA s RENAME TO s2"},
+		},
+		{
+			name: "qualified alter statements",
+			data: `
+CREATE DATABASE db;
+ALTER TABLE db.s.t ADD COLUMN col STRING;
+ALTER SCHEMA db.s RENAME TO s2;
+ALTER TABLE db.s.t ADD CONSTRAINT users_fk FOREIGN KEY (id) REFERENCES db.s.t2 (id) ON DELETE CASCADE;
+		`,
+			expectedStmts: []string{"ALTER TABLE crdb_temp_pgdump_import.s.t ADD COLUMN col STRING",
+				"ALTER SCHEMA crdb_temp_pgdump_import.s RENAME TO s2",
+				"ALTER TABLE crdb_temp_pgdump_import.s.t ADD CONSTRAINT users_fk FOREIGN KEY (id) REFERENCES crdb_temp_pgdump_import.s.t2 (id) ON DELETE CASCADE"},
+		},
+		{
+			name: "shp2pg.sql select statement",
+			data: `
+CREATE DATABASE db;
+CREATE TABLE "nyc_census_blocks" (gid serial,
+"blkid" varchar(15),
+"popn_total" float8,
+"popn_white" float8,
+"popn_black" float8,
+"popn_nativ" float8,
+"popn_asian" float8,
+"popn_other" float8,
+"boroname" varchar(32)) WITH (fillfactor = 2, autovacuum_enabled = false);
+ALTER TABLE "nyc_census_blocks" ADD PRIMARY KEY (gid);
+SELECT AddGeometryColumn('', 'nyc_census_blocks','geom','26918','MULTIPOLYGON',2);
+`,
+			expectedStmts: []string{"CREATE TABLE nyc_census_blocks (gid SERIAL8, blkid VARCHAR(15), popn_total FLOAT8, popn_white FLOAT8, popn_black FLOAT8, popn_nativ FLOAT8, popn_asian FLOAT8, popn_other FLOAT8, boroname VARCHAR(32))",
+				"ALTER TABLE nyc_census_blocks ADD PRIMARY KEY (gid)",
+				"ALTER TABLE nyc_census_blocks ADD COLUMN geom GEOMETRY(MULTIPOLYGON,26918)"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			data = test.data
+			h, err := parseDDLStatementsFromDumpFile(ctx, &evalCtx, execCtx, srv.URL,
```

Comment:
Could you add a comment explaining how the test's data is passed into `parseDDLStatmentsFromDumpFile`? I find it hard to figure out how srv.URL contains the test.data